front/src/modules/ui/data/data-table/table-cell/components/TableCellSoftFocusMode.tsx

```diff
@@ -27,34 +30,37 @@ export const TableCellSoftFocusMode = ({
   useScopedHotkeys(
     'enter',
     () => {
-      openTableCell();
+      if (!isFieldInputOnly) {
+        openTableCell();
+      } else {
+        // Maybe check for probability field
+        fieldRef?.current?.click();
```

Comment:
I think we shouldn't rely on automating user inputs here, instead we should stay focused on the minimal scope of our ticket : handling boolean fields only. 

So I see no problem creating custom logic for boolean type only where it's needed.

Preferably we would want to put this into an abstracted hook like useToggleEditOnlyInput and do the switch (or if else) statement in this hook with our isBoolean typeguard.

Reply:
Also we should avoid relying on refs here because there are many layers, and although it's a complex part of the app that must be managed with care, we shouldn't rely on shortcuts but instead take the time to understand the different layers to chose the right one to place or new logic.

Reply:
useToggleEditOnlyInput would also allow us to handle other types like probability you mentionned.

Reply:
Thank you for your review. I've implemented useToggleEditOnlyInput hook. But I have to say the previous one was not a shortcut but rather a way of imperative change the intrinsic state of component. I know there are not much in logic of reversing state but it's about the component only knows how to update itself.